packages/ckeditor5-table/src/tablewalker.ts

```diff
@@ -423,6 +432,67 @@ export default class TableWalker implements IterableIterator<TableSlot> {
 
 		rowSpans.set( column, data );
 	}
+
+	/**
+	 * Checks if part of the table can be skipped.
+	 */
+	private _canJumpToStartRow(): boolean {
+		return !!this._startRow &&
+			this._startRow > 0 &&
+			!this._updatedSpansBeforeStartRow;
+	}
+
+	/**
+	 * Sets the current row to `this._startRow` or the first row before it that has the number of cells
+	 * equal to the number of columns in the table.
+	 *
+	 * Example:
+	 * 	+----+----+----+
+	 *  | 00 | 01 | 02 |
+	 *  |----+----+----+
+	 *  | 10      | 12 |
+	 *  |         +----+
+	 *  |         | 22 |
+	 *  |         +----+
+	 *  |         | 32 | <--- Start row
+	 *  +----+----+----+
+	 *  | 40 | 41 | 42 |
+	 *  +----+----+----+
+	 *
+	 * If the 4th row is a `this._startRow`, this method will:
+	 * 1.) Count the number of columns this table has based on the first row (3 columns in this case).
+	 * 2.) Check if the 4th row contains 3 cells. It doesn't, so go to the row before it.
+	 * 3.) Check if the 3rd row contains 3 cells. It doesn't, so go to the row before it.
+	 * 4.) Check if the 2nd row contains 3 cells. It does, so set the current row to that row.
+	 *
+	 * Setting the current row this way is necessary to let the `next()`  method loop over the cells
+	 * spanning multiple rows or columns and update the `this._spannedCells` property.
+	 */
+	private _jumpToNonSpannedRowClosestToStartRow(): void {
+		const firstRow = this._table.getChild( 0 ) as Element;
+		const firstRowLength = this._getRowLenght( firstRow );
+
+		for ( let i = this._startRow!; i >= 0; i-- ) {
```

Comment:
Is it possible that the loop condition will stop the loop? Or it will always be terminated by `break`?
It should be, because the loop goes from some element in the direction to the first one and the first one has length equal to `firstRowLength`.
If so, what is the purpose of `i >= 0`?

Reply:
You are right, iterating over element `0` would compare it to itself. I changed it to `i > 0` and fixed the typo in method name.